kubespawner/objects.py

```diff
@@ -490,11 +490,17 @@ def make_ingress(
             V1beta1IngressBackend as ExtensionsV1beta1IngressBackend
         )
 
+    if routespec.startswith('/'):
+        host = None
+        path = routespec
+    else:
+        host, path = routespec.split('/', 1)
+
     meta = V1ObjectMeta(
         name=name,
         annotations={
             'hub.jupyter.org/proxy-data': json.dumps(data),
-            'hub.jupyter.org/proxy-routespec': routespec,
+            'hub.jupyter.org/proxy-routespec': path,
```

Comment:
what's the reason to make the proxy-routespec not contain the routespec anymore, and only the path component? I think this should probably stay as it is, but if the path and/or host should *also* be included, those should be additional fields.

Reply:
If we keep the full routespec here, it does not match the hub `_service_map` and get destroyed and recreated at each `check_routes` call.